odoo/addons/base/models/qweb.py

```diff
@@ -942,6 +942,9 @@ def _compile_directive_content(self, el, options, indent):
                 if isinstance(item, etree._Comment):
                     if self.env.context.get('preserve_comments'):
                         self._appendText("<!--%s-->" % item.text, options)
+                elif isinstance(item, etree._ProcessingInstruction):
+                    if self.env.context.get('preserve_comments'):
+                        self._appendText("<?%s %s?>" % (item.target, item.text), options)
```

Comment:
@rdeodoo I made this PR on 15.0, while the fix is actually for 13.0. In 13.0, there were no preserve_comments options, it was introduced in 15.0. And in 13.0, it's actually funny because this is the code here:
```python
# ignore comments & processing instructions
if isinstance(item, etree._Comment):
    continue
```
The code says it ignores comments **and processing instructions**... but it does not actually ignores processing instructions :shrug: (and the code was introduced like that)

My problem is, both in 13.0 and 15.0: it's not really stable to start ignoring ProcessingInstruction (even if they were not properly rendered as explained in the commit message)... so instead of reusing the `preserve_comments` param like I was planning too here, should I maybe introduce another "skip_processing_instructions" option ("skip" instead of "preserve" as the default would need to be false for stable policy) ?

... or maybe I am overthinking all of this again, this entire PR could just not be done and we let processing instruction being rendered as crappy stuff :shrug: (as the main fix at https://github.com/odoo/odoo/pull/92214 removes our own ProcessingInstruction when it normally should). But I'd rather fix those rendering :shrug:

Reply:
> or maybe I am overthinking all of this again, this entire PR could just not be done and we let processing instruction being rendered as crappy stuff

Why not just ignoring/hidding the ProcessingInstruction?

That would still address :point_down: 
> But I'd rather fix those rendering

Reply:
> Why not just ignoring/hidding the ProcessingInstruction?
> That would still address "But I'd rather fix those rendering"

Removing the rendering is not fixing it :thinking: What I meant is that if a processing instruction is added in the XML, there is no way to make it appear as intended in the final result.

We can go for that though but as explained, probably not in stable then? :grimacing:

Reply:
> Removing the rendering is not fixing it thinking

Why not? Not need to fix a problem if the problem is gone. At the end it is fixed :D 

> Maybe not in stable whatever the solution actually

I agree, let's not take the risk "for no reason" in stable, as the Odoo added processing instructions are correctly handled.
I guess you decide which one you prefer (delete or adapt to comment) and just retarget master?